task/bq2bq/executor/bumblebee/handler.py

```diff
@@ -8,20 +8,30 @@ class BigqueryJobHandler:
     def __init__(self) -> None:
         self._sum_slot_millis = 0
         self._sum_total_bytes_processed = 0
+        self.client = None
+        self.jobs = []
+        self._init_signal_handling()
+
+    def _init_signal_handling(self):
+        def handle_sigterm(signum, frame):
+            self._terminate_jobs()
+            sys.exit(1)
+        signal.signal(signal.SIGTERM, handle_sigterm)
+
+    def _terminate_jobs(self):
+        if self.client and self.jobs:
+            for job in self.jobs:
+                job_id = job.job_id
+                self.client.cancel_job(job_id)
+                logger.info(f"{job_id} successfully cancelled")
 
     def handle_job_finish(self, job) -> None:
         self._sum_slot_millis += job.slot_millis
         self._sum_total_bytes_processed += job.total_bytes_processed
 
     def handle_job_cancelled(self, client, job):
```

Comment:
rename the method as this does not handle cancellation. WDYT ?

Reply:
yes, this method would do register the job, maybe naming such `register_job_to_cancel` might be suitable? any suggestion?